src/Mvc/Mvc.ViewFeatures/test/Filters/AntiforgeryApplicationModelProviderTest.cs

```diff
@@ -0,0 +1,199 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System.Reflection;
+using Microsoft.AspNetCore.Antiforgery;
+using Microsoft.AspNetCore.Mvc.Core.Filters;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationModels;
+
+public class AntiforgeryApplicationModelProviderTest
```

Comment:
Should we have a case where the attributes are on the controller and the action? (For example, controller requires validation, and the action disables it)

Reply:
Good point! I've changed the code so that we only add the filter on the action if it hasn't been added on the controller. If there is a mismatch between the requiredness on a controller/action, the authorization filter will be added regardless but the middleware will not set an `IAntiforgeryValidationFeature` so the filter will effectively no-op.

We do incur a cost for running the filter on an action where it is not needed but I think that is fine to address in a follow-up.